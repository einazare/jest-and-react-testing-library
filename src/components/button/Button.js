import React from "react";
import "./Button.css";

function Button({ label }) {
  return (
    <div data-testid="button" className="button-style">
      Hi {label}
    </div>
  );
}

export default Button;
