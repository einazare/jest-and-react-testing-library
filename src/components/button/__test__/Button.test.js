import React from "react";
import ReactDOM from "react-dom";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom";
import renderer from "react-test-renderer";

import Button from "../Button.js";

afterEach(cleanup);

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Button></Button>, div);
});

it("renders button correctly 1", () => {
  const { getByTestId } = render(<Button label="Click me please"></Button>);
  expect(getByTestId("button")).toHaveTextContent("Click me please");
});

it("renders button correctly 2", () => {
  const { getByTestId } = render(<Button label="Save"></Button>);
  expect(getByTestId("button")).toHaveTextContent("Save");
});

it("matches snapshot 1", () => {
  const tree = renderer.create(<Button label="Save"></Button>).toJSON();
  expect(tree).toMatchSnapshot();
});

it("matches snapshot 2", () => {
  const tree = renderer
    .create(<Button label="Click me please"></Button>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
