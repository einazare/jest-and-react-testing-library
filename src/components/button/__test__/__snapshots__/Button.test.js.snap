// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1 1`] = `
<div
  className="button-style"
  data-testid="button"
>
  Hi 
  Save
</div>
`;

exports[`matches snapshot 2 1`] = `
<div
  className="button-style"
  data-testid="button"
>
  Hi 
  Click me please
</div>
`;
